File: app/src/main/java/br/com/digitalhouse/projetomarvel/network/MarvelAPI.java
package br.com.digitalhouse.projetomarvel.network;

import br.com.digitalhouse.projetomarvel.pojo.ComicsResponse;
import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface MarvelAPI {

    @GET("comics")
    Observable<ComicsResponse> ComicsResponse(
            @Query("format") String format,
            @Query("formatType") String formatType,
            @Query("noVariants") boolean noVariants,
            @Query("ts") String ts,
            @Query("hash") String hash,
            @Query("apikey") String apikey);
}

File: app/src/main/java/br/com/digitalhouse/projetomarvel/network/MarvelApi.kt
package br.com.digitalhouse.projetomarvel.network

import br.com.digitalhouse.projetomarvel.api.ComicsResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query

interface MarvelAPI {
    @GET("comics")
    fun getAllCharacters(
            @Query("format") format: String?,
            @Query("formatType") formatType: String?,
            @Query("noVariants") noVariants: Boolean,
            @Query("ts") ts: String?,
            @Query("hash") hash: String?,
            @Query("apikey") apikey: String?,
            @Query("offset") offset: Int? = 0): Observable<ComicsResponse?>?


}

